### PR TITLE
Pdr 269 refresh token

### PR DESCRIPTION
**Pull request by Nahuel-Salvatierra**
# Summary

- Implementation of an endpoint to refresh user sessions.

# Detail

A new endpoint was implemented to refresh user sessions in the system. During this implementation, an issue arose while loading fixtures. The solution involved removing the `cascade:true` property in the relationship with the user entity.

Additionally, unnecessary files were identified and successfully removed from the repository to streamline and clean up the codebase.

# Reference

In the context of the implementation and issue resolution, the TypeORM documentation on [relationships and cascades](https://typeorm.io/relations#cascades) was consulted for detailed information on managing cascades in entity relationships.

### CODE REVIEW

File: src/module/auth/application/service/auth.service.ts [hunk 1]
```diff
@@ -148,21 +147,23 @@ export class AuthService {
...
+      });
+
+      if (verify) {
+        const user = await this.userService.getUserByEmail(verify.email);
+        const accessToken = await this.getRefreshToken(user);
+        const newAuth = new Auth(accessToken, user);
+        await this.authRepository.saveRefreshToken(newAuth);
+        return accessToken;
+      }
+    } catch (error) {
```
Comment (Fludeo) [inline src/module/auth/application/service/auth.service.ts:170]: I think this try-catch is not necessary, this code came from the express app, In the express app was needed

File: src/module/auth/interface/auth.controller.ts [hunk 2]
```diff
@@ -39,17 +39,17 @@ export class AuthController {
...
+    if (cookie === undefined) {
+      throw new HttpException('Access denied', 403);
+    }
+    const httpOnlyToken: string = cookie?.split('=')[1].split(';')[0];
+    const newAccessToken = await this.authService.refreshToken(httpOnlyToken);
 
-  //   res.status(200);
-  //   res.json({ access_token: newAccessToken });
-  // }
+    res.status(200);
```
Comment (Fludeo) [inline src/module/auth/interface/auth.controller.ts]: Remember we are using nest now. We should not be using res i guess. ( Correct me if I'm wrong )

**Review by Fludeo — CHANGES_REQUESTED** [review]

File: src/module/auth/application/service/auth.service.ts (hunk 1 above)
Comment (Nahuel-Salvatierra) [reply src/module/auth/application/service/auth.service.ts:170]: I have been reviewing how to remove this try-catch, but JWT throws a token verification error when receiving invalid tokens. Now, I am handling that error from the catch block. I can't seem to find the class to which this error belongs to throw a more appropriate error.

File: src/module/auth/interface/auth.controller.ts (hunk 2 above)
Comment (Nahuel-Salvatierra) [reply src/module/auth/interface/auth.controller.ts]: When I use the return option, the tests indicate that error responses are handled, but when I need the correct response, the response time expires, indicating that the server is not responding. Should there be a way to include the @Res in the return?

**Review by Fludeo — APPROVED** [review]

**Pull request merged**